### PR TITLE
SearchV3 on mobile chat

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers This adds the searchv3 functionality in mobile.

I changed the feature flag to have it enabled for `__DEV__`. We don't currently have an admin thing on native, so we may want that soon so people can play with this.

### CODE REVIEW

File: shared/chat/conversation/index.native.js [hunk 1]
```diff
@@ -30,30 +32,47 @@ const Conversation = (props: Props) => (
...
-    />
-    <Banner />
-    {props.showLoader && <LoadingLine />}
-    {props.finalizeInfo
-      ? <OldProfileResetNotice />
-      : <Input
-          focusInputCounter={props.focusInputCounter}
-          onEditLastMessage={props.onEditLastMessage}
-          onScrollDown={props.onScrollDown}
+    {props.inSearch
```
Comment (chrisnojima) [inline shared/chat/conversation/index.native.js]: maybe we should break these out into their own constants or something
const Header = props.inSearch ? ...
then just {Header} here

File: shared/chat/search.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      id: id,
+      followingState: SearchConstants.followStateHelper(state, username, service),
+      // $FlowIssue ??
+      icon: serviceIdToIcon(serviceId),
+      username,
+      service,
+    }
+  })
+
+  console.log('items', userItems)
```
Comment (chrisnojima) [inline shared/chat/search.js]: 🔥

File: shared/chat/search.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import {compose, withState, defaultProps, withHandlers} from 'recompose'
+import {connect} from 'react-redux'
+import {globalStyles, globalMargins} from '../styles'
+import {parseUserId, serviceIdToIcon} from '../util/platforms'
+import {chatSearchResultArray} from '../constants/selectors'
+import {onChangeSelectedSearchResultHoc} from '../searchv3/helpers'
+
+import type {TypedState} from '../constants/reducer'
+
+const mapStateToProps = (state: TypedState) => {
```
Comment (chrisnojima) [inline shared/chat/search.js]: we should use recompose here else this'll always fail the connect cache and always rerender right

Comment (chrisnojima) [inline shared/chat/search.js]: we dont need to do this now per-se but we might want to put a comment / ticket

**Comment by chrisnojima** [comment]
also is this still compatible with searchv1 or does this make it only searchv3

**Comment by keybase-ci-visdiff** [comment]
The commits 99294fe44a04...43748bbeec12 introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498237641617-99294fe44a04-43748bbeec12-linux/Tracker-New_user_Scroll_1-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498237641617-99294fe44a04-43748bbeec12-linux/Tracker-New_user_Scroll_1-43748bbeec12.png)  
   Tracker-New_user_Scroll_1 comparison
 * Changed: **Search_V_3_results_List-keybase_Results_One** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498237641617-99294fe44a04-43748bbeec12-linux/Search_V_3_results_List-keybase_Results_One-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498237641617-99294fe44a04-43748bbeec12-linux/Search_V_3_results_List-keybase_Results_One-43748bbeec12.png)  
   Search_V_3_results_List-keybase_Results_One comparison
 * Changed: **Search_V_3_results_List-keybase_Results** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498237641617-99294fe44a04-43748bbeec12-linux/Search_V_3_results_List-keybase_Results-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498237641617-99294fe44a04-43748bbeec12-linux/Search_V_3_results_List-keybase_Results-43748bbeec12.png)  
   Search_V_3_results_List-keybase_Results comparison
 * Changed: **Search_V_3_results_List-facebook_Results** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498237641617-99294fe44a04-43748bbeec12-linux/Search_V_3_results_List-facebook_Results-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498237641617-99294fe44a04-43748bbeec12-linux/Search_V_3_results_List-facebook_Results-43748bbeec12.png)  
   Search_V_3_results_List-facebook_Results comparison
</summary>

File: shared/chat/conversation/header-or-search-header.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// @flow
+import React from 'react'
+import Header from './header/container'
+import SearchHeader from '../search-header'
+import * as SearchConstants from '../../constants/searchv3'
+
+type Props = {
```
Comment (chrisnojima) [inline shared/chat/conversation/header-or-search-header.js]: i was thinking just even in that file was fine, but breaking it out is fine too

File: shared/chat/search.js (hunk 3 above)
Comment (MarcoPolo) [inline shared/chat/search.js]: done

File: shared/chat/conversation/index.native.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/chat/conversation/index.native.js]: I made a component that can be used on the desktop version too

**Comment by MarcoPolo** [comment]
> also is this still compatible with searchv1 or does this make it only searchv3

Still compatible. The only weird thing is if you are creating a chat and are still in the pending/list building state. Then go to the search tab and start a conversation through the searchv1 flow. You'll be in the temporary conversation with a search header (because you're still in searchv3 - the store's inSearch value is true).

**Comment by keybase-ci-visdiff** [comment]
The commits 99294fe44a04...ac5a993e596b introduce visual changes on linux. <details><summary>:mag_right: 3 changed</summary>

 * Changed: **Search_V_3_results_List-keybase_Results_One** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498238907186-99294fe44a04-ac5a993e596b-linux/Search_V_3_results_List-keybase_Results_One-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498238907186-99294fe44a04-ac5a993e596b-linux/Search_V_3_results_List-keybase_Results_One-ac5a993e596b.png)  
   Search_V_3_results_List-keybase_Results_One comparison
 * Changed: **Search_V_3_results_List-keybase_Results** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498238907186-99294fe44a04-ac5a993e596b-linux/Search_V_3_results_List-keybase_Results-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498238907186-99294fe44a04-ac5a993e596b-linux/Search_V_3_results_List-keybase_Results-ac5a993e596b.png)  
   Search_V_3_results_List-keybase_Results comparison
 * Changed: **Search_V_3_results_List-facebook_Results** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498238907186-99294fe44a04-ac5a993e596b-linux/Search_V_3_results_List-facebook_Results-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498238907186-99294fe44a04-ac5a993e596b-linux/Search_V_3_results_List-facebook_Results-ac5a993e596b.png)  
   Search_V_3_results_List-facebook_Results comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 99294fe44a04...f711214d890d introduce visual changes on linux. <details><summary>:mag_right: 4 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498245522701-99294fe44a04-f711214d890d-linux/Tracker-New_user_Scroll_1-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498245522701-99294fe44a04-f711214d890d-linux/Tracker-New_user_Scroll_1-f711214d890d.png)  
   Tracker-New_user_Scroll_1 comparison
 * Changed: **Search_V_3_results_List-keybase_Results_One** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498245522701-99294fe44a04-f711214d890d-linux/Search_V_3_results_List-keybase_Results_One-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498245522701-99294fe44a04-f711214d890d-linux/Search_V_3_results_List-keybase_Results_One-f711214d890d.png)  
   Search_V_3_results_List-keybase_Results_One comparison
 * Changed: **Search_V_3_results_List-keybase_Results** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498245522701-99294fe44a04-f711214d890d-linux/Search_V_3_results_List-keybase_Results-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498245522701-99294fe44a04-f711214d890d-linux/Search_V_3_results_List-keybase_Results-f711214d890d.png)  
   Search_V_3_results_List-keybase_Results comparison
 * Changed: **Search_V_3_results_List-facebook_Results** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498245522701-99294fe44a04-f711214d890d-linux/Search_V_3_results_List-facebook_Results-99294fe44a04.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1498245522701-99294fe44a04-f711214d890d-linux/Search_V_3_results_List-facebook_Results-f711214d890d.png)  
   Search_V_3_results_List-facebook_Results comparison
</summary>

**Pull request merged**